src/NATS.Client.Core/Commands/PingCommand.cs

```diff
@@ -1,68 +1,8 @@
-using NATS.Client.Core.Internal;
-
 namespace NATS.Client.Core.Commands;
 
-internal sealed class PingCommand : CommandBase<PingCommand>
+internal sealed class PingCommand
 {
-    private PingCommand()
-    {
-    }
-
-    public static PingCommand Create(ObjectPool pool, CancellationTimer timer)
-    {
-        if (!TryRent(pool, out var result))
-        {
-            result = new PingCommand();
-        }
-
-        result.SetCancellationTimer(timer);
-
-        return result;
-    }
-
-    public override void Write(ProtocolWriter writer)
-    {
-        writer.WritePing();
-    }
-
-    protected override void Reset()
-    {
-    }
-}
-
-internal sealed class AsyncPingCommand : AsyncCommandBase<AsyncPingCommand, TimeSpan>
-{
-    private NatsConnection? _connection;
-
-    private AsyncPingCommand()
-    {
-    }
-
-    public DateTimeOffset? WriteTime { get; private set; }
-
-    public static AsyncPingCommand Create(NatsConnection connection, ObjectPool pool, CancellationTimer timer)
-    {
-        if (!TryRent(pool, out var result))
-        {
-            result = new AsyncPingCommand();
-        }
-
-        result._connection = connection;
-        result.SetCancellationTimer(timer);
-
-        return result;
-    }
-
-    public override void Write(ProtocolWriter writer)
-    {
-        WriteTime = DateTimeOffset.UtcNow;
-        _connection!.EnqueuePing(this);
-        writer.WritePing();
-    }
+    public DateTimeOffset WriteTime { get; } = DateTimeOffset.UtcNow;
 
-    protected override void Reset()
-    {
-        WriteTime = null;
-        _connection = null;
-    }
+    public TaskCompletionSource<TimeSpan> TaskCompletionSource { get; } = new(TaskCreationOptions.RunContinuationsAsynchronously);
```

Comment:
If we do this It would be nice to have a github issue to try to pool in future (since I'd assume ping is easy to do that with)

Reply:
Absolutely, if this merges we should make a follow-up to pool this in the future

Reply:
see #321